osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmgmt/application/endpoints/SecretManagementEndpoint.java

```diff
@@ -0,0 +1,172 @@
+import org.opensmartgridplatform.schemas.security.secretmanagement._2020._05.SecretTypes;
+import org.opensmartgridplatform.schemas.security.secretmanagement._2020._05.StoreSecretsRequest;
+import org.opensmartgridplatform.schemas.security.secretmanagement._2020._05.StoreSecretsResponse;
+import org.opensmartgridplatform.schemas.security.secretmanagement._2020._05.TechnicalFault;
+import org.opensmartgridplatform.schemas.security.secretmanagement._2020._05.TypedSecrets;
+import org.opensmartgridplatform.secretmgmt.application.domain.SecretType;
+import org.opensmartgridplatform.secretmgmt.application.domain.TypedSecret;
+import org.opensmartgridplatform.secretmgmt.application.exception.TechnicalServiceFaultException;
+import org.opensmartgridplatform.secretmgmt.application.services.SecretManagementService;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.EncryptedSecret;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.Secret;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.EncryptionProvider;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.EncryptionProviderType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ws.server.endpoint.annotation.Endpoint;
+import org.springframework.ws.server.endpoint.annotation.PayloadRoot;
+import org.springframework.ws.server.endpoint.annotation.RequestPayload;
+import org.springframework.ws.server.endpoint.annotation.ResponsePayload;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensmartgridplatform.secretmgmt.application.config.ApplicationConfig.COMPONENT_NAME;
+
+@Endpoint
+public class SecretManagementEndpoint {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecretManagementEndpoint.class);
+
+    private static final String NAMESPACE_URI = "http://www.opensmartgridplatform.org/schemas/security/secretmanagement/2020/05";
+    private static final String KEY_REFERENCE = "1"; //only one key in use
+
+    private final SecretManagementService secretManagementService;
+    private final EncryptionProvider jreEncryptionProvider;
+
+    public SecretManagementEndpoint(SecretManagementService secretManagementService, EncryptionProvider soapSecretEncryptionProvider) {
+        this.secretManagementService = secretManagementService;
+        this.jreEncryptionProvider = soapSecretEncryptionProvider;
+    }
+
+    @PayloadRoot(namespace = NAMESPACE_URI, localPart = "getSecretsRequest")
+    @ResponsePayload
+    public GetSecretsResponse getSecretsRequest(@RequestPayload GetSecretsRequest request) {
+
+        LOGGER.info("Handling incoming SOAP request 'getSecretsRequest' for device {}", request.getDeviceId());
+
+        GetSecretsResponse response = new GetSecretsResponse();
+
+        try {
+            SecretTypes soapSecretTypes = request.getSecretTypes();
```

Comment:
for an Endpoint, this method contains too much code, consider moving this to a service for better readability, maintenance and testability